agent/consul/connect_ca_endpoint.go

```diff
@@ -183,6 +183,23 @@ func (s *ConnectCA) ConfigurationSet(
 		return fmt.Errorf("error generating CA root certificate: %v", err)
 	}
 
+	commonConfig, err := config.GetCommonConfig()
+	if err != nil {
+		return err
+	}
+	if commonConfig.LeafCertTTL < newProvider.MinimumLeafTTL() {
+		if commonConfig.ForceWithoutCrossSigning {
+			s.srv.logger.Printf("[INFO] configured leaf TTL of %v is less than provider minimum of %v. "+
+				"ForceWithoutCrossSigning is set, continuing with operation",
+				commonConfig.LeafCertTTL, newProvider.MinimumLeafTTL())
+		} else {
+			return fmt.Errorf("configured leaf TTL of %v is less than provider minimum of %v. "+
+				"Use the ForceWithoutCrossSigning parameter to force it anyway",
+				commonConfig.LeafCertTTL, newProvider.MinimumLeafTTL())
+		}
```

Comment:
It doesn't make sense to me that we relax the validation of some other TTL config when `-force-without-cross-signing` is given. The only purpose of that flag is to indicate that it's OK that the new CA won't be cross signed by the old one (because it can't be for some reason) which may cause temporary connection issues during rotation roll out.

I think we should still hard fail bad configuration for all other reasons.